antarest/study/business/binding_constraint_management.py

```diff
@@ -239,11 +250,26 @@ def get_binding_constraint(
             binding_constraint.append(new_config)
         return binding_constraint
 
+    def validate_binding_constraint(self, study: Study, constraint_id: str) -> None:
+        if int(study.version) < 870:
+            return  # There's nothing to check for constraints before v8.7
+        file_study = self.storage_service.get_storage(study).get_raw(study)
+        config = file_study.tree.get(["input", "bindingconstraints", "bindingconstraints"])
+        group = next((value["group"] for value in config.values() if value["id"] == constraint_id), None)
```

Comment:
It's impossible for a bc after v8.7 to not have a group so this case will never happen.
Therefore, it the group is None, it means the bc does not exist, that's why i raised such an Exception

Reply:
In fact it's possible if you update a BC via the `/raw` endpoint you can set the group to `None`. Which in this case raise a key error instead of the one you want. (Maybe it's a edge case, just to let you know about that)

Reply:
You're right but it would be a non-sense.